homeassistant/components/livebox/sensor.py

```diff
@@ -0,0 +1,210 @@
+"""Sensor for Livebox router."""
+import logging
+from datetime import timedelta
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice,
+    DEVICE_CLASS_CONNECTIVITY,
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+from . import DOMAIN
+from .const import TEMPLATE_SENSOR
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(minutes=5)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the sensors."""
+    box = hass.data[DOMAIN]
+    async_add_entities(
+        [
+            RXSensor(box, config_entry),
+            TXSensor(box, config_entry),
+            InfoSensor(box, config_entry),
+        ],
+        True,
+    )
+
+
+class LiveboxSensor(Entity):
+    """Representation of a livebox sensor."""
+
+    _name = "generic"
+
+    def __init__(self, box, config_entry):
+        """Initialize the sensor."""
+
+        self._box = box
+        self._box_id = config_entry.data["box_id"]
+        self._state = None
+        self._datas = None
```

Comment:
This instance attribute isn't used except as a local variable. We should remove it and just use a local variable in the update methods.